pkg/addonmanager/manager.go

```diff
@@ -321,12 +345,50 @@ func (a *addonManager) StartWithInformers(ctx context.Context,
 	return nil
 }
 
-// New returns a new Manager for creating addon agents.
-func New(config *rest.Config) (AddonManager, error) {
-	return &addonManager{
+// ManagerOptions defines the flags for addon manager
```

Comment:
can we put this Option flag in the separated dir specifically for cloudevent?

Reply:
put this to separate cloudevents package.